tests/resolver/test_filtering.py

```diff
@@ -0,0 +1,405 @@
+                vdescs=vdescs,
+                allowed_namespaces=[],
+            )
+        )
+        == []
+    )
+
+    # Non existing namespace allowed - should return empty list
+    assert (
+        list(
+            filter_variants_by_namespaces(
+                vdescs=vdescs,
+                allowed_namespaces=["NonExistentNamespace"],
+            )
+        )
+        == []
+    )
+
+    # Only `OmniCorp` allowed - should return `vdesc1`
+    assert list(
+        filter_variants_by_namespaces(
+            vdescs=vdescs,
+            allowed_namespaces=["OmniCorp"],
+        )
+    ) == [vdesc1]
+
+    # Only `TyrellCorporation` allowed - should return `vdesc2`
+    assert list(
+        filter_variants_by_namespaces(
+            vdescs=vdescs,
+            allowed_namespaces=["TyrellCorporation"],
+        )
+    ) == [vdesc2]
+
+    # Both `OmniCorp` and `TyrellCorporation` - should return all `vdescs`
+    # Note: order should not matter
+    assert (
+        list(
+            filter_variants_by_namespaces(
+                vdescs=vdescs,
+                allowed_namespaces=["OmniCorp", "TyrellCorporation"],
+            )
+        )
+        == vdescs
+    )
+
+    assert (
+        list(
+            filter_variants_by_namespaces(
+                vdescs=vdescs,
+                allowed_namespaces=["TyrellCorporation", "OmniCorp"],
+            )
+        )
+        == vdescs
+    )
+
+
+def test_filter_variants_by_namespaces_validation_error(
+    vdescs: list[VariantDescription],
+):
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_namespaces(
+                vdescs=vdescs,
+                allowed_namespaces="not a list",  # type: ignore[arg-type]
+            ),
+            maxlen=0,
+        )
+
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_namespaces(
+                vdescs=vdescs,
+                allowed_namespaces=[1234],  # type: ignore[list-item]
+            ),
+            maxlen=0,
+        )
+
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_namespaces(
+                vdescs="not a list",  # type: ignore[arg-type]
+                allowed_namespaces=["OmniCorp"],
+            ),
+            maxlen=0,
+        )
+
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_namespaces(
+                vdescs=["not a `VariantDescription`"],  # type: ignore[list-item]
+                allowed_namespaces=["OmniCorp"],
+            ),
+            maxlen=0,
+        )
+
+
+# ====================== `filter_variants_by_features` ====================== #
+
+
+def test_filter_variants_by_features(vdescs: list[VariantDescription]):
+    vdesc1, vdesc2, _ = vdescs
+
+    vfeat1 = VariantFeature(namespace="OmniCorp", feature="custom_feat")
+    vfeat2 = VariantFeature(namespace="TyrellCorporation", feature="client_id")
```

Comment:
I wonder if we should a helper to convert `VariantProperty` into a `VariantFeature`. In #32 I've inlined similar logic as well.

Reply:
I actually added it : https://github.com/wheelnext/variantlib/pull/25/files#diff-2739a87bc9b043288bb679e59c1a2200798497f3d166f9e8fdea8d0620cce6d7R119-R122